tests/test_public_exports.py

```diff
@@ -0,0 +1,31 @@
+import fastapi
+
+
+def test_fastapi_module_exports():
```

Comment:
```suggestion
def test_fastapi_module_exports():
"""
Test the publicly exported module set.
Benefits for declaring explicitly the public exports:
    * Using __all__ sets what gets imported with `from fastapi import *`
    * Enables mypy to pass all checks with fastapi when using `implicit_reexport = False` 
"""
import yourpackage

missing = set(n for n in yourpackage.__all__
              if getattr(yourpackage, n, None) is None)
self.assertEmpty(missing, msg=f"__all__ contains unresolved names: {missing}")
```
Reason: You're not checking if something else is being imported. 
Reference: http://xion.io/post/code/python-all-wild-imports.html

Reply:
I don't know how to suggest a whole block of code, sorry :(

Reply:
this is a valid point. I'll add it a bit later. I also have to work on some submodule __all__ definitions